src/compiler/checker.ts

```diff
@@ -6609,7 +6611,7 @@ module ts {
             }
 
             // The identityMapper object is used to indicate that function expressions are wildcards
-            if (contextualMapper === identityMapper) {
+            if (contextualMapper === identityMapper && isContextSensitive(node)) {

```

Comment:
I see. In other words, we only passed the identity mapper during the argument assignability check. And because that is a boolean check whose only side effects apply to context sensitive functions, it was impossible to observe the fact that context insensitive functions  were being skipped.